db/seeds.rb

```diff
@@ -5,3 +5,108 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+
+require 'csv'
+
+# seed one user, if he/she doesn't exist
+def seed_user
+  user = User.find_by(email: 'user1@email.com')
+  if user
+    puts "Dummy user has already been added."
+  else
+    user = User.create!(name: 'user1', email: 'user1@email.com', password: 'password')
+    user.interests.push(Category.find_by(subcategory_name: "Arcades"))
+    user.interests.push(Category.find_by(subcategory_name: "Music Venues"))
+    user.save!
+    puts "Added dummy user '#{user.name}' with email '#{user.email}' to Users."
+  end
+end
+
+# Pick a route
+
+def seed_locations
+  # seed cities
+  CSV.foreach('db/major-us-cities.csv', headers: true) do |row|
+    longlat = { type: "Point", coordinates: [row['longitude'], row['latitude']] }
+    Location.find_or_create_by(city: row['city'],
+                             region: row['region'],
+                            longlat: longlat)
+  end
+
+  # seed parks
+  # TODO
+end
+
+def seed_categories
+  CSV.foreach('db/yelp-categories.csv', headers: true) do |row|
+    Category.find_or_create_by(name: row['name'],
+                               code: row['code'],
+                   subcategory_code: row['subcategory_code'],
+                   subcategory_name: row['subcategory_name'])
+  end
+end
+
+def has_location?(business)
+  business.respond_to?(:location) && business.location.respond_to?(:coordinate)
+end
+
+def has_categories?(business)
+  business.respond_to?(:categories)
+end
+
+def has_website?(business)
+  business.respond_to?(:url)
+end
+
+def seed_attractions
+  Location.all[0..10].each do |location|
+    Category.each do |category|
+
+      # check if you have already requested Yelp for this location and category
+      unless Seed.find_by(location: location, category: category)
+        puts "Searching for #{category.subcategory_code} in #{location.city}, #{location.region}"
+
+        result = Yelp.client.search_by_coordinates(
```

Comment:
This is *a lot* of logic to have in your seed file, especially logic that is not tested.  This is a good use case for a service object, such as `AttractionImporter` to handle searching yelp and importing those things into the database.  You also might consider then adding logic that checks if there already is data for that category and skipping it, which would allow you to incrementally add more categories without having to rebuild the database.